cloudigrade/api/tests/clouds/azure/tasks/maintenance/test_repopulate_azure_instance_definitions.py

```diff
@@ -0,0 +1,114 @@
+"""Collection of tests for api.tasks.cloudtrail.repopulate_azure_instance_mapping."""
+import json
+from unittest.mock import MagicMock, Mock, patch
+
+import faker
+from django.test import TestCase
+
+from api.clouds.azure import tasks
+from api.models import InstanceDefinition
+
+_faker = faker.Faker()
+
+# Token for this string since some of the paths in "patch" uses are very long.
+_MAINTENANCE = "api.clouds.azure.tasks.maintenance"
+
+
+class RepopulateAzureInstanceMappingTest(TestCase):
+    """Celery task 'repopulate_azure_instance_mapping' test cases."""
+
+    def setUp(self):
+        """Set up common variables for tests."""
+        self.standard_a5_definition = Mock()
+        self.standard_a5_definition.name = "Standard_A5"
+        self.standard_a5_definition.resource_type = "virtualMachines"
+        vcpu_capability = Mock()
+        vcpu_capability.name = "vCPUs"
+        vcpu_capability.value = 2
+        memory_capability = Mock()
+        memory_capability.name = "MemoryGB"
+        memory_capability.value = 14
+        capabilities_list = [vcpu_capability, memory_capability]
+        self.standard_a5_definition.capabilities = capabilities_list
+        self.good_json = json.dumps(
+            self.standard_a5_definition, default=lambda x: x.__dict__
+        )
+
+        self.bad_standard_a5_definition = Mock()
+        self.bad_standard_a5_definition.name = "Standard_A5"
+        self.bad_standard_a5_definition.resource_type = "virtualMachines"
+        vcpu_capability = Mock()
+        vcpu_capability.name = "vCPUs"
+        vcpu_capability.value = 1.21
+        memory_capability = Mock()
+        memory_capability.name = "MemoryGB"
+        memory_capability.value = 41
+        capabilities_list = [vcpu_capability, memory_capability]
+        self.bad_standard_a5_definition.capabilities = capabilities_list
+        self.bad_json = json.dumps(
+            self.bad_standard_a5_definition, default=lambda x: x.__dict__
+        )
+
+        self.garbage_definition = Mock()
+        self.garbage_definition.name = "RandomAzureObject"
+        self.garbage_definition.resource_type = "notVirtualMachines"
+
+    @patch("api.util.InstanceDefinition.objects.get_or_create")
+    @patch(f"{_MAINTENANCE}.ComputeManagementClient")
+    def test_repopulate_azure_instance_mapping(self, mock_compute, mock_db_create):
```

Comment:
Why did you choose to mock out the interaction with the `InstanceDefinition` model? In most (all?) of our other tests, we let the DB write happen, and we query to see that it was written correctly. Is there something different about this particular test interaction that I might not be seeing?

Reply:
No other reason than copying the pattern from `test_repopulate_ec2_instance_mapping.py`. I could rewrite both tests to remove the mock.

Reply:
Huh. Not sure why we did that originally. 🤔 

Yeah, I think letting this go through the real model classes is the better path.